File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03147/p03147_num4_parsed.cpp
#include <iostream>
#include <vector>

int minimizeWateringOperations(int N, std::vector<int>& heights) {
    int operations = 0;
    
    for (int i = 1; i <= N; i++) {
        int diff = heights[i] - heights[i - 1];
        if (diff > 0) {
            operations += diff;
            heights[i] -= diff;
        }
    }
    
    return operations;
}

int main() {
    int N;
    std::cin >> N;
    
    std::vector<int> heights(N + 1);
    for (int i = 1; i <= N; i++) {
        std::cin >> heights[i];
    }
    
    int minOperations = minimizeWateringOperations(N, heights);
    
    std::cout << minOperations << std::endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03147/p03147_num4_parsed.py
# Function to calculate minimum number of watering operations required

def calculate_min_watering_operations(N, heights):
    operations = 0
    
    # Iterate over each flower
    for k in range(N):
        # Increase the height of the flower
        operations += heights[k]
        
        # If the height of the next flower is greater than the current flower
        if k < N-1 and heights[k+1] > heights[k]:
            # Increase the height of the current flower by the difference
            operations += heights[k+1] - heights[k]
    
    return operations

# Read input from standard input
N = int(input())
heights = list(map(int, input().split()))

# Calculate minimum number of watering operations
min_operations = calculate_min_watering_operations(N, heights)

# Print the result
print(min_operations)